### ISSUE TITLE
Enable markdown in comment forms

### ISSUE THREAD

**Issue by nfagerlund** [dreamwidth/dw-free#2573]
Today, DW has three markup dialects for entries and comments:

- HTML with auto-linebreaks (default)
- HTML without auto-linebreaks
- Markdown

(I'm eliding some complexity here, because there are multiple interacting properties that produce all this. But that's the observable behavior.) 

You can switch to raw HTML for comments, but only if you're replying from the full talkform. And you can't use Markdown in the comment forms at all. But the machinery to deal with Markdown comments is already here! Reply-by-email comments actually get saved and displayed as Markdown.

So, the comment forms should let you choose your markup dialect. 

I already know how to do this, but:

- ~This should wait until after #2540, because that might change the underpinnings of the `editor` prop.~
- Should also wait for #664, which might influence the desired design.

Once this is up and running, should make another ticket for editor helper buttons, to simplify basic markup like bold, italic, lists, links, images, etc. That's another one that will be heavily influenced by #664.
